src/state.rs

```diff
@@ -250,3 +261,49 @@ impl Stack {
         self.stack[index] = value;
     }
 }
+
+impl Default for Heap {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Heap {
+    pub fn new() -> Self {
+        Self { heap: vec![] }
+    }
+    // Returns how many ergs the expand costs
+    pub fn expand_memory(&mut self, address: u32) -> u32 {
+        if address >= self.heap.len() as u32 {
+            let old_size = self.heap.len() as u32;
+            self.heap.resize(address as usize + 1, 0);
+            return MEMORY_GROWTH_ERGS_PER_BYTE * (address - old_size + 1);
+        }
+        0
+    }
+
+    pub fn store(&mut self, address: u32, value: U256) {
+        for i in 0..32 {
+            self.heap[address as usize + i] = value.byte(31 - i);
```

Comment:
Any reason why we aren't using .to_big_endian and simply extending the vector with it? The spec states this is interpreted as big endian. Page 189, if you need further reference.

Reply:
I didn't realize that, however, the best thing I can come up with is
```
let mut bytes: [u8;32] = [0;32];
value.to_big_endian(&mut bytes);
for i in 0..32 {
      self.heap[address as usize + i] = bytes[i];
}
```
I can't just extend the vector, since there is an address were it should be stored

Reply:
Yup